kyo-core/shared/src/main/scala/kyo/layers.scala

```diff
@@ -0,0 +1,131 @@
+package kyo
+
+object layers {
+
+  trait Layer[In, Out] { self =>
+    def run[T, S](effect: T > (S with In))(implicit fl: Flat[T > (S with In)]): T > (S with Out)
+
+    final def add[Out1, In1](other: Layer[In1, Out1]): Layer[In with In1, Out with Out1] =
```

Comment:
Could you rename to `andThen`? I think it'd be better to communicate that order is important.